addons/account_facturx/models/account_move.py

```diff
@@ -232,12 +232,19 @@ def _import_facturx_invoice(self, tree):
                         # Price Unit.
                         line_elements = element.xpath('.//ram:GrossPriceProductTradePrice/ram:ChargeAmount', namespaces=tree.nsmap)
                         if line_elements:
-                            invoice_line_form.price_unit = float(line_elements[0].text) / invoice_line_form.quantity
+                            invoice_line_form.price_unit = float(line_elements[0].text)
+
+                            line_elements = element.xpath('.//ram:GrossPriceProductTradePrice/ram:BasisQuantity', namespaces=tree.nsmap)
+                            if line_elements:
+                                invoice_line_form.price_unit /= float(line_elements[0].text)
                         else:
                             line_elements = element.xpath('.//ram:NetPriceProductTradePrice/ram:ChargeAmount', namespaces=tree.nsmap)
                             if line_elements:
-                                invoice_line_form.price_unit = float(line_elements[0].text) / invoice_line_form.quantity
+                                invoice_line_form.price_unit = float(line_elements[0].text)
 
+                                line_elements = element.xpath('.//ram:NetPriceProductTradePrice/ram:BasisQuantity', namespaces=tree.nsmap)
+                                if line_elements:
+                                    invoice_line_form.price_unit /= float(line_elements[0].text)
```

Comment:
don't made multiple assignation because there are triggering `onchange` and then, you are loosing performances.

Reply:
Thanks, I've fixed that